packages/ipfs-core/src/components/block/put.js

```diff
@@ -50,8 +64,8 @@ module.exports = ({ blockService, pin, gcLock, preload }) => {
         const mhtype = options.mhtype || 'sha2-256'
         const format = options.format || 'dag-pb'
 
-        /** @type {CidVersion} */
-        let cidVersion
+        /** @type {CIDVersion} */
+        let cidVersion = 1
```

Comment:
I'm afraid answer is bit complicated.  You are right that it is assigned but TS seems to not recognize that here and complains that `cidVersion` maybe undefined.

Use of ternary instead of `if else`  would have worked, but then it end up nesting and ESLint complained about it. So this seemed like a reasonable way to keep everyone happy 🤷‍♂️